sys/arduino/base.cpp

```diff
@@ -58,3 +61,40 @@ void delay(unsigned long msec)
 {
     xtimer_usleep(1000 * msec);
 }
+
+#ifdef ADC_NUMOF
+int analogRead(int adc_pin)
+{
+    int adc_value;
+
+    /* State of ADC channel.
+     * 0: Not initialized
+     * 1: Successfully initialized
+     */
+    static int adc_state[ADC_NUMOF];
+
+    /* Check if the ADC line is valid */
+    if ((adc_pin < 0) || (adc_pin >= ADC_NUMOF)) {
+        return -1;
+    }
+
+    /* Initialization of given ADC channel */
+    if (adc_state[adc_pin] == 0) {
+        if (adc_init(ADC_LINE(adc_pin)) == 0) {
```

Comment:
The mapping of the RIOT ADC lines to the Arduino ADC pins is in the adc_init() funktion. Therefore, I only adapted the Arduino pin numbers to the ADC lines through a subtraction of ADC_PIN_OFFSET.